core/internal/runfiles/uploader.go

```diff
@@ -157,6 +183,91 @@ func (u *uploader) lockForOperation(method string) bool {
 	return true
 }
 
+// Begins watching the given path and uploading when the file changes.
+func (u *uploader) watch(path string) error {
+	// Lazily start the watcher when we receive our first file to watch.
+	if u.watcherOrNil == nil {
+		if err := u.startWatcher(); err != nil {
+			return err
+		}
+	}
+
+	if err := u.watcherOrNil.Add(path); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Starts up the file watcher goroutine.
+func (u *uploader) startWatcher() error {
+	if u.watcherOrNil != nil {
+		return fmt.Errorf(
+			"runfiles: tried to start watcher, but it is already started",
+		)
+	}
+
+	u.watcherOrNil = watcher.New()
+	u.watcherOrNil.FilterOps(watcher.Write)
+
+	grp, ctx := errgroup.WithContext(context.Background())
+	u.watcherWG.Add(2)
+
+	grp.Go(func() error {
+		defer u.watcherWG.Done()
+
+		u.loopWatchFiles(ctx)
+
+		return nil
+	})
+
+	grp.Go(func() error {
+		defer u.watcherWG.Done()
+
+		if err := u.watcherOrNil.Start(time.Millisecond * 500); err != nil {
```

Comment:
it's kinda silly, but should we have this value as constant and potential in the future configurable or just hard code it here and that's all?

Reply:
done, extracted to a constant! it's a good idea here because otherwise it's unnamed and you have to check the docs for Start()